borg/helpers.py

```diff
@@ -777,30 +796,56 @@ def parse(self, text):
-            self.host = m.group('host')
-            self.port = m.group('port') and int(m.group('port')) or None
-            self.path = os.path.normpath(m.group('path'))
-            self.archive = m.group('archive')
-            return True
-        m = self.file_re.match(text)
-        if m:
-            self.proto = m.group('proto')
-            self.path = os.path.normpath(m.group('path'))
-            self.archive = m.group('archive')
-            return True
-        m = self.scp_re.match(text)
-        if m:
-            self.user = m.group('user')
-            self.host = m.group('host')
-            self.path = os.path.normpath(m.group('path'))
-            self.archive = m.group('archive')
-            self.proto = self.host and 'ssh' or 'file'
-            return True
-        return False
+        if sys.platform != 'win32':
+            m = self.ssh_re.match(text)
+            if m:
+                self.proto = m.group('proto')
+                self.user = m.group('user')
+                self.host = m.group('host')
+                self.port = m.group('port') and int(m.group('port')) or None
+                self.path = os.path.normpath(m.group('path'))
+                self.archive = m.group('archive')
+                return True
+            m = self.file_re.match(text)
+            if m:
+                self.proto = m.group('proto')
+                self.path = os.path.normpath(m.group('path'))
+                self.archive = m.group('archive')
+                return True
+            m = self.scp_re.match(text)
+            if m:
+                self.user = m.group('user')
+                self.host = m.group('host')
+                self.path = os.path.normpath(m.group('path'))
+                self.archive = m.group('archive')
+                self.proto = self.host and 'ssh' or 'file'
+                return True
+            return False
+        else:
+            m = self.file_re.match(text)
+            if m:
+                self.proto = m.group('proto')
+                self.path = os.path.normpath(m.group('drive') + ":\\" + m.group('path'))
+                self.archive = m.group('archive')
+                return True
+            m = self.ssh_re.match(text)
+            if m:
+                self.proto = m.group('proto')
+                self.user = m.group('user')
+                self.host = m.group('host')
+                self.port = m.group('port') and int(m.group('port')) or None
+                self.path = os.path.normpath(m.group('path'))
+                self.archive = m.group('archive')
+                return True
+            m = self.scp_re.match(text)
+            if m:
+                self.user = m.group('user')
+                self.host = m.group('host')
+                self.path = os.path.normpath(m.group('path'))
+                self.archive = m.group('archive')
+                self.proto = self.host and 'ssh' or 'file'
+                return True
+            return False
```

Comment:
The regular expression needs to be different on Windows since it must accept `D:/path/to/repo`, but that path must be not accepted on other platforms.